src/main/java/org/broadinstitute/hellbender/utils/collections/IntervalsSkipListOneContig.java

```diff
@@ -51,10 +53,15 @@ public IntervalsSkipListOneContig(Iterable<T> loc) {
         // heuristic: if we have too many small buckets then we're better off instead
         // taking fewer but bigger steps, and then iterating through a few values.
         // Thus, put a lower bound on bucket size.
-        if (bSize<32) bSize=32;
+        if (bSize < 32) {
+            bSize = 32;
+        }
         shift = floorLog2(bSize);
-        sortVariants();
-        buildIndexAndCheck();
+
+        vs.sort(Comparator.comparing(Locatable::getContig).thenComparingInt(Locatable::getStart));
```

Comment:
Should this sort also sort by `getEnd`? It might reduce some weirdness at the slight expense of additional sorting cost.